src/v/cluster/members_manager.cc

```diff
@@ -206,15 +206,16 @@ wait_for_next_join_retry(std::chrono::milliseconds tout, ss::abort_source& as) {
 ss::future<result<join_reply>> members_manager::dispatch_join_to_remote(
   const config::seed_server& target, model::broker joining_node) {
     vlog(clusterlog.info, "Sending join request to {}", target.addr);
-
+    auto timeout = rpc::clock_type::now() + _join_timeout;
```

Comment:
this looks like a duration to me. 

question: did you see how seastar does the RPC timeout? i think they use durations.

Reply:
right, what we care about is this shouldn't take more than `3seconds` or whatever, which is not a timepoint. it is a duration. Internally we need to turn the duration *into* a timepoint, but I think the *api* should use a duration.... maybe overload? (i doubt the latter since we don't have any `deadline` type workloads)

Reply:
I've checked the Seastar RPC api. Timeouts in seastar RPC are expressed as `time_points`. For me it was always easier and more natural to use duration than a time point. I think that the timeout however has sense in this case. Our RPC timeout is now expressed as a time point not duration. secondly if we want something to be limited in time for example take up to 5 seconds we have to take into account both the connection plus request timeout. With time point it is easy to reason about when request will finish, no matter if it is connection or processing timeout.